web/src/composebox_typeahead.ts

```diff
@@ -996,6 +999,57 @@ export function content_highlighter_html(item: TypeaheadSuggestion): string | un
     }
 }
 
+function will_produce_broken_stream_topic_link(word: string): boolean {
+    return invalid_stream_topic_regex.test(word);
+}
+
+function get_stream_name_from_topic_link_syntax(syntax: string): string {
+    const start = syntax.lastIndexOf("#**");
+    const end = syntax.lastIndexOf(">");
+    return syntax.slice(start + 3, end);
+}
+
+export function html_escape_markdown_syntax_characters(text: string): string {
+    return text.replaceAll(invalid_stream_topic_regex, (match) => {
+        switch (match) {
+            case "`":
+                return "&grave;";
+            case ">":
+                return "&gt;";
```

Comment:
Why does this have `>`, when `will_produce_broken_stream_topic_link` does not? Seems like a comment should document this detail.

Reply:
I had planned to include `>` as an invalid character for stream or topic name, but doing so does not cover all the cases when `>` can lead to a broken link. 
Things work fine if we select a stream from the typeahead containing `>` in the name, i.e, a fallback markdown syntax is generated.
But if we then select a topic, such that the final syntax is something like `#**stream>name>topicname`, a broken link is generated (since we can't know which `>` is for the syntax and which is a part of stream/topic name).

I am adding `>` in the regex for now, since it does cover some cases.

Reply:
A related problem is that we can't `topic_jump` when a fallback stream link is generated. (initiated a discussion [here](https://chat.zulip.org/#narrow/stream/101-design/topic/Avoid.20generation.20broken.20stream-topic.20links)).

Reply:
That seems reasonable.
I would say it is better than the current behaviour, since it at least handles stream links for stream names with problematic characters.

@timabbott What is the next step for this PR?